proto/proto/wallet.proto

```diff
@@ -68,90 +39,56 @@ message StatusResult {
     uint64 sync_height = 3;
 }
 
-// TODO: Placeholder transaction response, replace as needed
-message TransactionResult {}
-
-// Addresses
-message ShowAddress {        
-    // The index of the address to show.
-    u32 index = 1;
-    // If true, emits only the address and not the (local) label for it.
-    bool addr_only = 2;
-}
-message CreateNewAddress {
-    // A freeform label for the address, stored only locally.
-    string label = 1;
+// A note plaintext with associated metadata about its status.
+message NoteRecord {
+    // The note commitment, identifying the note.
+    crypto.NoteCommitment note_commitment = 1;
+    // The note plaintext itself.
+    crypto.Note note = 2;
+    // A precomputed decryption of the note's diversifier index.
+    crypto.DiversifierIndex diversifier_index = 3;
+    // The note's nullifier.
+    crypto.Nullifier nullifier = 4;
+    // The height at which the note was created.
+    uint64 height_created = 5;
+    // Records whether the note was spent (and if so, at what height).
+    optional uint64 height_spent = 6;
```

Comment:
The `NoteRecord` type in #667 should turn into a domain type for this proto type.